src/graphnet/data/extractors/i3genericextractor.py

```diff
@@ -0,0 +1,205 @@
+from typing import Any, Dict, List, Optional, Union
+
+from graphnet.data.extractors.i3extractor import I3Extractor
+from graphnet.data.extractors.utilities.types import (
+    cast_object_to_pure_python,
+    cast_pulse_series_to_pure_python,
+)
+from graphnet.data.extractors.utilities.collections import (
+    transpose_list_of_dicts,
+    serialise,
+    flatten_nested_dictionary,
+)
+
+from graphnet.utilities.logging import get_logger
+
+logger = get_logger()
+
+try:
+    from icecube import (
+        dataclasses,
+        icetray,
+    )  # pyright: reportMissingImports=false
+except ImportError:
+    logger.warning("icecube package not available.")
+
+
+GENERIC_EXTRACTOR_NAME = "<GENERIC>"
+
+
+class I3GenericExtractor(I3Extractor):
+    """Dynamically and generically extract all information from frames."""
```

Comment:
Last we spoke, there were indeed a few keys we could not extract generically. If this is still the case, it should be clear from this docstring which fields we leave unextracted.